quickwit-doc-mapper/src/query_builder.rs

```diff
@@ -82,41 +84,97 @@ fn has_range_clause(user_input_ast: &UserInputAst) -> bool {
 /// Tells if the query has a Term or Range node which does not
 /// specify a search field.
 fn needs_default_search_field(user_input_ast: &UserInputAst) -> bool {
+    collect_field_nodes(user_input_ast)
+        .iter()
+        .any(|field_opt| field_opt.is_none())
+}
+
+fn resolve_fields(schema: &Schema, field_names: &[String]) -> anyhow::Result<Vec<Field>> {
+    let mut fields = vec![];
+    for field_name in field_names {
+        let field = schema
+            .get_field(field_name)
+            .ok_or_else(|| TantivyQueryParserError::FieldDoesNotExist(field_name.clone()))?;
+        fields.push(field);
+    }
+    Ok(fields)
+}
+
+// Collects the fields nodes on term or range ast nodes
+fn collect_field_nodes(user_input_ast: &UserInputAst) -> Vec<Option<String>> {
```

Comment:
the name `collect_field_nodes` is confusing, does it collect field nodes or field names?